### PR TITLE
feat(workflow): add upsert memo command

### LINKED ISSUE #1124: [Feature Request] Add `upsertMemo` command
### Describe the solution you'd like

- Add `upsertMemo` command to Workflow API.

### Additional context

- Underlying support already exists in Core SDK.

### PR DESCRIPTION
**Pull request by tharun208**
## What was changed

Implement `upsertMemo` command

## Why?
   See https://github.com/temporalio/features/issues/119

## Checklist

1. Closes #1124 

2. How was this tested:
Added integration tests 

3. Any docs updates needed?

### CODE REVIEW

**Comment by bergundy** [comment]
Thanks!

Overall lgtm.

Please also test deleting a memo and reading the memo from WorkflowHandle.describe, I think setting a value to null or undefined should work.

For testing, use the new style at https://github.com/temporalio/sdk-typescript/blob/main/packages/test/src/test-integration-workflows.ts

File: packages/test/src/integration-tests-old.ts [hunk 1]
```diff
@@ -677,6 +677,20 @@ export function runIntegrationTests(codec?: PayloadCodec): void {
     );
   });
 
+  test('Workflow can upsert memo', async (t) => {
+    const { client } = t.context;
+    const workflow = await client.start(workflows.upsertAndReadMemo, {
+      taskQueue: 'test',
+      workflowId: uuid4(),
+      args: [{ note2: 'bar' }],
```
Comment (bergundy) [inline packages/test/src/integration-tests-old.ts:685]: Please use the newer `test-integration-workflows.ts` and test adding, updating and removing.

**Review by bergundy — COMMENTED** [review]
Thanks!

LGTM apart from the testing part.
LMK if you need help with that.

**Review by intelligent002-rewire — APPROVED** [review]

**Comment by mjameswh** [comment]
@tharun208 Do you want to continue working on this, or should we take it on? We'd need some more tests, as pointed out by bergundy previously.

**Comment by tharun208** [comment]
Hi @mjameswh was on PTO, but will look into this week

**Comment by nhalstead** [comment]
Any update on this getting merged?

**Comment by kencrim** [comment]
Bump on the ETA for this? Would be happy to pitch in!

**Comment by mjameswh** [comment]
>Hi, I will work on it today

@tharun208 Ah! I missed your message. I just fixed your PR.

File: packages/test/src/workflows/upsert-and-read-memo.ts
Comment (bergundy) [inline packages/test/src/workflows/upsert-and-read-memo.ts:1]: Can you find a cleaner way to test sinks without duplicating the workflow?

File: packages/workflow/src/workflow.ts [hunk 2]
```diff
@@ -1366,6 +1366,75 @@ export function upsertSearchAttributes(searchAttributes: SearchAttributes): void
...
+ * }
+ * ```
+ *
+ * @param memo The Record to merge.
+ */
+export function upsertMemo(memo: Record<string, unknown>): void {
+  const activator = assertInWorkflowContext('Workflow.upsertMemo(...) may only be used from a Workflow Execution.');
+
+  if (memo == null) {
+    throw new Error('memo must be a non-null Record');
```
Comment (bergundy) [inline packages/workflow/src/workflow.ts:1409]:
```suggestion
    throw new TypeError('memo must be a non-null Record');
```

**Review by bergundy — APPROVED** [review]

**Review by mjameswh — APPROVED** [review]

**Pull request merged**